cosine_similarity_model.py

```diff
@@ -0,0 +1,141 @@
+import joblib
+import logging
+import numpy as np
+from pymongo import MongoClient
+import pandas as pd
+from sklearn.metrics.pairwise import cosine_similarity
+from sklearn.feature_extraction.text import CountVectorizer
+
+
+class MovieRecommender:
```

Comment:
split the training and recommendation functionality into separate classes
recommendation class will load pre trained model and compute similarity scores for a given input
here is simple workflow:

1. use the `MovieTrainer` class to preprocess data and train the model
2. save the trained model to disk
3. use the `MovieRecommender` class to load the trained model and generate recommendations

also move the training class and recommendation class into separate files